### PR TITLE
[4.0] TFA forms not shown in frontend

### PR DESCRIPTION
**Pull request by SharkyKZ**
Fixes https://github.com/joomla/joomla-cms/issues/29300.

### Summary of Changes

Fixes two factor authentication forms not being shown in frontend.
Fixes selected TFA method form not shown in backend (probably).
Removes duplicate JS.

### Testing Instructions

Enable some two factor authentication plugins.
Edit profile in frontend.
Select TFA method in the dropdown.

### Expected result

TFA setup form shown.

### Actual result

TFA setup form not shown.

### Documentation Changes Required

No.

### CODE REVIEW

File: components/com_users/tmpl/profile/edit.php [hunk 1]
```diff
@@ -22,7 +22,7 @@
 $lang = Factory::getLanguage();
 $lang->load('plg_user_profile', JPATH_ADMINISTRATOR);
 
-HTMLHelper::_('script', 'com_users/two-factor-switcher.min.js', array('version' => 'auto', 'relative' => true));
+HTMLHelper::_('script', 'com_users/two-factor-switcher.js', array('version' => 'auto', 'relative' => true));
```
Comment (dgrammatiko) [inline components/com_users/tmpl/profile/edit.php]: please change this to:
```php
HTMLHelper::_('script', 'com_users/two-factor-switcher.js', ['version' => 'auto', 'relative' => true], ['defer' => true]);
```

File: administrator/components/com_users/tmpl/user/edit.php [hunk 2]
```diff
@@ -18,7 +18,7 @@
 
 HTMLHelper::_('behavior.formvalidator');
 HTMLHelper::_('behavior.keepalive');
-HTMLHelper::_('script', 'com_users/admin-users-user.min.js', array('version' => 'auto', 'relative' => true));
+HTMLHelper::_('script', 'com_users/two-factor-switcher.js', array('version' => 'auto', 'relative' => true));
```
Comment (dgrammatiko) [inline administrator/components/com_users/tmpl/user/edit.php]: please change this to 
```php
HTMLHelper::_('script', 'com_users/two-factor-switcher.js', ['version' => 'auto', 'relative' => true], ['defer' => true]);
```

File: administrator/components/com_users/tmpl/user/edit.php [hunk 3]
```diff
@@ -71,14 +70,15 @@
...
-			<?php echo HTMLHelper::_('select.genericlist', Usershelper::getTwoFactorMethods(), 'jform[twofactor][method]', array('onchange' => 'Joomla.twoFactorMethodChange()', 'class' => 'custom-select'), 'value', 'text', $this->otpConfig->method, 'jform_twofactor_method', false); ?>
+			<?php echo HTMLHelper::_('select.genericlist', UsersHelper::getTwoFactorMethods(), 'jform[twofactor][method]', array('onchange' => 'Joomla.twoFactorMethodChange();', 'class' => 'custom-select'), 'value', 'text', $this->otpConfig->method, 'jform_twofactor_method', false); ?>
 		</div>
 	</div>
 	<div id="com_users_twofactor_forms_container">
 		<?php foreach ($this->tfaform as $form) : ?>
-		<div id="com_users_twofactor_<?php echo $form['method'] ?>" class="hidden">
-			<?php echo $form['form'] ?>
-		</div>
+			<?php $class = $form['method'] == $this->otpConfig->method ? '' : 'class="hidden"'; ?>
```
Comment (Quy) [inline administrator/components/com_users/tmpl/user/edit.php]:
```suggestion
			<?php $class = $form['method'] == $this->otpConfig->method ? '' : ' class="hidden"'; ?>
```

File: components/com_users/tmpl/profile/edit.php [hunk 4]
```diff
@@ -82,13 +82,13 @@
...
 					<div class="controls">
-						<?php echo HTMLHelper::_('select.genericlist', $this->twofactormethods, 'jform[twofactor][method]', array('onchange' => 'Joomla.twoFactorMethodChange()'), 'value', 'text', $this->otpConfig->method, 'jform_twofactor_method', false); ?>
+						<?php echo HTMLHelper::_('select.genericlist', $this->twofactormethods, 'jform[twofactor][method]', array('onchange' => 'Joomla.twoFactorMethodChange();', 'class' => 'custom-select'), 'value', 'text', $this->otpConfig->method, 'jform_twofactor_method', false); ?>
 					</div>
 				</div>
 				<div id="com_users_twofactor_forms_container" class="com-users-profile__twofactor-form">
 					<?php foreach ($this->twofactorform as $form) : ?>
-						<?php $style = $form['method'] == $this->otpConfig->method ? 'display: block' : 'display: none'; ?>
-						<div id="com_users_twofactor_<?php echo $form['method']; ?>" style="<?php echo $style; ?>">
+						<?php $class = $form['method'] == $this->otpConfig->method ? '' : 'class="hidden"'; ?>
```
Comment (Quy) [inline components/com_users/tmpl/profile/edit.php]:
```suggestion
						<?php $class = $form['method'] == $this->otpConfig->method ? '' : ' class="hidden"'; ?>
```

**Comment by richard67** [comment]
Added release blocker label as inherited from the issue.

**Comment by richard67** [comment]
I have tested this item :white_check_mark: successfully on 544672318ac40d16f8557f917cb4b97c089417d4<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/29365">issues.joomla.org/tracker/joomla-cms/29365</a>.</sub>

**Comment by bonzani** [comment]
I have tested this item :white_check_mark: successfully on 544672318ac40d16f8557f917cb4b97c089417d4<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/29365">issues.joomla.org/tracker/joomla-cms/29365</a>.</sub>

**Comment by richard67** [comment]
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/29365">issues.joomla.org/tracker/joomla-cms/29365</a>.</sub>

**Comment by wilsonge** [comment]
Thanks!

**Pull request merged**